### ISSUE TITLE
Having trouble seeing a difference between boto3's batch_writer and this package

### ISSUE THREAD

**Issue by nitrocode** [aio-libs/aiobotocore#572]
I have a script that uses `boto3`

        with open(file_name, 'r') as items:
            with table.batch_writer() as batch:
                for idx, item in enumerate(items):
                    item = item.strip()
                    batch.put_item(
                        Item={
                            'item': item,
                        }
                    )

and another that uses `aiobotocore`

       with open(file_name, 'r') as items:
            batch = []
            count = 0
            for item in items:
                item = item.strip()
                batch.append(item)
                if len(batch) == size:
                    count += size
                    response = await self.client.batch_write_item(
                         RequestItems={
                             table_name: [
                                 {'PutRequest': {'Item': {
                                     'item': {
                                          'S': item,
                                      }
                                 }}}
                                 for ditem in batch
                             ]
                         }
                    )
                    batch = []

When I run either, I get about 64 to 70 seconds to finish roughly 15k records.

What am I doing wrong? I tried to follow the example [dynamodb_batch_write](https://github.com/aio-libs/aiobotocore/blob/master/examples/dynamodb_batch_write.py) as closely as possible.

**Comment by thehesiod** [comment]
I'm not sure what the question is, aiobotocore doesn't make AWS calls magically faster :)

**Comment by nitrocode** [comment]
I guess I tried aiobotocore in order to make my calls asynchronous and in doing so I was hoping that I could get higher throughput. I'm curious of course on how I can improve my throughput but a better question is what exactly does aiobotocore provide me that boto3 does not provide me that could help me improve my throughput? And if it cannot improve my throughput, what benefits does aiobotocore give over boto3?

**Comment by jettify** [comment]
In your script you perform sequential upload, one butch after other. In order to have higher throughput, try to upload data with multiple coroutines. For example you do not have to wait for the first batch to complete in order to start uploading new batch.

**Comment by thehesiod** [comment]
ya, @nitrocode I think you need to read about what asyncio is. It allows you to run multiple tasks in parallel, it will not make individual tasks any faster.

**Comment by nitrocode** [comment]
@thehesiod yes I've used asyncio before and I probably wrongly assumed that this package, the way I wrote it above, would batch all the PutRequests at once like you coded in your example. 

@jettify I suppose I should just use asyncio instead of aiobotocore if I want to run multiple batches simultaneously.

Please correct me if I'm misunderstanding something and thank you for all the comments.

**Comment by thehesiod** [comment]
@nitrocode aiobotocore is simply an async version of botocore with an aiohttp backend instead of requests, no extra fanciness :)  Ya you could theoretically use botocore  + a bunch `loop.run_in_executor` if you take care of the threading correctly.  However if you want a true async client use this.
